data/bookkeeping/txn_merkle.go

```diff
@@ -81,10 +93,13 @@ func txnMerkleToRaw(txid [crypto.DigestSize]byte, stib [crypto.DigestSize]byte)
 func (tme *txnMerkleElem) ToBeHashed() (protocol.HashID, []byte) {
 	// The leaf contains two hashes: the transaction ID (hash of the
 	// transaction itself), and the hash of the entire SignedTxnInBlock.
-	txid := tme.txn.ID()
-	stib := tme.stib.Hash()
-
-	return protocol.TxnMerkleLeaf, txnMerkleToRaw(txid, stib)
+	if tme.hashType == crypto.Sha512_256 {
+		stib := tme.stib.Hash()
+		return protocol.TxnMerkleLeaf, txnMerkleToRaw(tme.txn.ID(), stib)
+	}
+	// else: hashType == crypto.Sha256
+	stib := tme.stib.HashSHA256()
```

Comment:
nit: please make consistent (the same) logic for the `tme.hashType` selection here and below.

Reply:
fixed here [`81f49d3` (#3829)](https://github.com/algorand/go-algorand/pull/3829/commits/81f49d33fa2521d0d06545986da46bd12101f03c)